pkg/data/values_test.go

```diff
@@ -0,0 +1,135 @@
+package data
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetValue(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name        string
+		data        interface{}
+		keys        []string
+		wantValue   interface{}
+		wantSuccess bool
+	}{
+		{
+			name:        "nil map",
+			data:        nil,
+			keys:        []string{"somekey"},
+			wantValue:   nil,
+			wantSuccess: false,
+		},
+		{
+			name: "key is not in map",
+			data: map[string]interface{}{
+				"realKey": "realVal",
+			},
+			keys:        []string{"badKey"},
+			wantValue:   nil,
+			wantSuccess: false,
+		},
+		{
+			name: "key is in first level of map",
+			data: map[string]interface{}{
+				"realKey": "realVal",
+			},
+			keys:        []string{"realKey"},
+			wantValue:   "realVal",
+			wantSuccess: true,
+		},
+		{
+			name: "key is nested in map",
+			data: map[string]interface{}{
+				"parent": map[string]interface{}{
+					"child": map[string]interface{}{
+						"grandchild": "someValue",
+					},
+				},
+			},
+			keys:        []string{"parent", "child", "grandchild"},
+			wantValue:   "someValue",
+			wantSuccess: true,
+		},
+		{
+			name: "incorrected nested key",
+			data: map[string]interface{}{
+				"parent": map[string]interface{}{
+					"child": map[string]interface{}{
+						"grandchild": "someValue",
+					},
+				},
+			},
+			keys:        []string{"parent", "grandchild", "child"},
+			wantValue:   nil,
+			wantSuccess: false,
+		},
+		{
+			name: "get index of slice",
+			data: map[string]interface{}{
+				"parent": map[string]interface{}{
+					"children": []interface{}{
+						"alice",
+						"bob",
+						"eve",
+					},
+				},
+			},
+			keys:        []string{"parent", "children", "2"},
+			wantValue:   "eve",
+			wantSuccess: true,
+		},
+		{
+			name: "get index of top levelslice",
```

Comment:
```suggestion
			name: "get index of top level slice",
```

What about a case where there is a slice of maps with each map having a key whose value is a slice of elements? I think it might be good to add...

Reply:
I've adjusted the name of this test, added the requested test case, and added another test case which attempted to use a string as a slice index value. Please re-open if this doesn't meet your suggestion here.